tests/match/test_match.py

```diff
@@ -137,18 +102,38 @@ def test_rand_SGM(self):
         score = chr12c.score_
         assert 11156 <= score < 12500
 
-        A, B = self._get_AB("chr15a")
-        chr15a = self.rand.fit(A, B)
-        score = chr15a.score_
-        assert 9896 <= score < 12000
-
-        n = A.shape[0]
-        pi = np.array([5, 10, 8, 13, 12, 11, 14, 2, 4, 6, 7, 15, 3, 1, 9]) - [1] * n
-        W1 = [0, 5, 11, 14]
-        W2 = [pi[z] for z in W1]
-        chr15a = self.rand.fit(A, B, W1, W2)
-        score = chr15a.score_
-        assert 9896 <= score < 10000
+    def test_padding(self):
+        directed = False
+        loops = False
+        block_probs = [
+            [0.9, 0.4, 0.3, 0.2],
+            [0.4, 0.9, 0.4, 0.3],
+            [0.3, 0.4, 0.9, 0.4],
+            [0.2, 0.3, 0.4, 0.7],
+        ]
+        n = 100
+        n_blocks = 4
+        rho = 0.5
+        block_members = np.array(n_blocks * [n])
+        n_verts = block_members.sum()
+        G1p, G2 = sbm_corr(block_members, block_probs, rho, directed, loops)
+        G1 = np.zeros((300, 300))
+
+        for i in range(5):
+            stepx = 100 * i
+            for j in range(5):
+                stepy = 100 * j
+                G1[(75 * i) : (75 * (i + 1)), (75 * j) : (75 * (j + 1))] = G1p[
+                    stepx : (stepx + 75), stepy : (stepy + 75)
+                ]
+        seed1 = np.random.randint(0, 300, 10)
+        seed2 = [int(x / 75) * 25 + x for x in seed1]
+        res = self.barygm.fit(G2, G1, seed2, seed1)
+        matching = np.concatenate(
+            [res.perm_inds_[x * 100 : (x * 100) + 75] for x in range(n_blocks)]
+        )
+
+        assert 1.0 == (sum(matching == np.arange(300)) / 300)
 
```

Comment:
i have a pretty hard time reading this test at a glance. 

could you add some comments explaining what is going on here, or better yet, just make the code simpler? 

I'm wondering if we also need this extensive/specific of a test? seems like super small rho-ER  with rho very high or 1, delete 1 or 2 vertices, show that the matching still works for the vertices that were not deleted in either graph. or something along those lines. not sure if we'd need one for adopted and one for naive, or maybe on a super simple example the same padding scheme would get the same answer? but we could test both